sdk/ai/ai-projects/src/agents/utils.ts

```diff
@@ -40,17 +40,154 @@ export enum connectionToolType {
+    definition : {
+      type: toolType,
+      [toolType]: {
+        connections: connectionIds.map(connectionId => ({connection_id: connectionId}))
+      }
     }
   }
 }
+
+/**
+ * Creates a file search tool
+ *
+ * @param vectorStoreIds - The ID of the vector store attached to this agent. There can be a maximum of 1 vector store attached to the agent.
+ * @param vectorStores - The list of vector store configuration objects from Azure. This list is limited to one element. The only element of this list contains the list of azure asset IDs used by the search tool.
+ * @param definitionDetails - The input definition information for a file search tool as used to configure an agent.
+ * 
+ * @returns An object containing the definition and resources for the file search tool
+ */
+export function createFileSearchTool(
+  vectorStoreIds?: string[],
+  vectorStores?: Array<VectorStoreConfigurations>,
+  definitionDetails? : FileSearchToolDefinitionDetails
+) : { definition : FileSearchToolDefinition, resources: ToolResources} {
+  return {
+    definition : { type: "file_search", file_search: definitionDetails },
+    resources : { file_search: { vector_store_ids: vectorStoreIds, vector_stores: vectorStores} }
+  }
+}
+
+/**
+ * Creates a code interpreter tool
+ * 
+ * @param fileIds - A list of file IDs made available to the `code_interpreter` tool. There can be a maximum of 20 files associated with the tool.
+ * @param dataSources - The data sources to be used. This option is mutually exclusive with fileIds.
+ * 
+ * @returns An object containing the definition and resources for the code interpreter tool.
+ */
+export function createCodeInterpreterTool( 
+  fileIds?: string[],
+  dataSources?: Array<VectorStoreDataSource>
+) : { definition : CodeInterpreterToolDefinition, resources: ToolResources} {
+  if (fileIds && dataSources) {
+    throw new Error("Cannot specify both fileIds and dataSources");
+  }
+  
+  return {
+    definition : { type: "code_interpreter" },
+    resources : { code_interpreter: { file_ids: fileIds, data_sources: dataSources } }
+  }
+}
+
+/**
+ * Creates an Azure AI search tool
+ * 
+ * @param indexConnectionId - The connection ID of the Azure AI search index.
+ * @param indexName - The name of the Azure AI search index.
+ * 
+ * @returns An object containing the definition and resources for the Azure AI search tool.
+ */
+export function createAzureAISearchTool( 
+  indexConnectionId: string,
+  indexName: string
+) : { definition : AzureAISearchToolDefinition, resources: ToolResources} {
+  return {
+    definition : { type: "azure_ai_search" },
+    resources : { azure_ai_search: { indexes: [{ index_connection_id: indexConnectionId, index_name: indexName  }] } }
+  }
+}
+
+/**
+ * Represents a set of tools with their definitions and resources.
+ */
+export class ToolSet {
+  /** A list of tool definitions that have been added to the tool set. */
```

Comment:
suggestion tools:{ defintion:ToolDefintion, resource:ToolResouce}[]

Reply:
I think we want to keep definitions and resources separate so that when the user creates an agent they can easily do so like this: `client.agents.createAgent("gpt4", {tools: toolSet.definition, tools_resources: toolSet.resources})`